hw3/hw3-react/src/search/SearchPage.js

```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react';
+import SearchBar from './SearchBar';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Figure from 'react-bootstrap/Figure';
+import FigureImage from 'react-bootstrap/FigureImage';
+import FigureCaption from 'react-bootstrap/FigureCaption';
+
+function SearchPage() {
+  const [results, setResults] = useState([]);
+  const characterMatches = results.map((character) => (
+    <Col>
+      <Figure id={`img-${character.id}`} className="bg-white p-2">
+        <FigureImage
+          width={'250px'}
+          height={'300px'}
+          src={character.imageUrl}
+          alt={`A portrait of ${character.fullName}`}
+        ></FigureImage>
+        <FigureCaption>{character.fullName}</FigureCaption>
+        <small className="d-block">{character.title}</small>
+      </Figure>
+    </Col>
+  ));
+
+  function searchName(name) {
+    // this isn't working
+    setResults([]);
+    alert(`You searched for ${name}!`);
+    const storedCharacters = sessionStorage.getItem('characters');
+    const characters = JSON.parse(storedCharacters);
+    console.log(characters.data);
+
+    characters.data.map((character) => {
+      if (name.toLowerCase() === character.fullName.toLowerCase()) {
+        alert(`There was a match for ${name}!`);
+        setResults([...results, character]);
```

Comment:
Search function ALMOST works, the logic is basically there! Once you get a match on a character, storing that character's information in an object and using setResult should work. Then in your return statement, doing a conditional check on if the object exist, displaying that object's properties.

Reply:
Your feedback was really helpful in getting me to think about what setResults() really does. In my refactor I used `array.filter()` instead and removed the `...results` in `setResults` to clear any previous matches.